core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1931,20 +1920,6 @@ private[spark] object Utils
     path
   }
 
-  /**
-   * Updates Spark config with properties from a set of Properties.
-   * Provided properties have the highest priority.
-   */
-  def updateSparkConfigFromProperties(
```

Comment:
The first time PR is here:
https://github.com/apache/spark/pull/14650/files#diff-2ecc6aef4b0c50bbf146e6c0b3b8b2249375f06a83e2a224c7718cfc850c3af7R1969

In `Spark 4.0`, `mesos` is no longer supported and the relevant code has been removed.